packages/utils/src/collections.ts

```diff
@@ -22,3 +22,17 @@ export function zip3<A, B, C>(as: A[], bs: B[], cs: C[]) {
 export function notEmpty<TValue>(value: TValue | null | undefined): value is TValue {
   return value !== null && value !== undefined
 }
+
+// https://stackoverflow.com/questions/33355528/filtering-an-array-with-a-function-that-returns-a-promise
+export function mapAsync<T, U>(
```

Comment:
there's `concurrentMap` for this. As a rule of thumb, we should be aware of running an unbounded amount of concurrent functions. There's always some bound to consider.

Reply:
how should I think about the bound? to me it seems like any number of `call()`s can be safely concurrent